hazelcast-client/src/main/java/com/hazelcast/client/config/YamlClientFailoverConfigBuilder.java

```diff
@@ -0,0 +1,162 @@
+     * it begins with 'classpath:', then a classpath resource is loaded. Else it will assume it is a file
+     * reference</li>
+     * <li>it checks if a hazelcast-client-failover.yaml is available in the working dir</li>
+     * <li>it checks if a hazelcast-client-failover.yaml is available on the classpath</li>
+     * </ol>
+     * @throws HazelcastException if no failover configuration is found
+     */
+    public YamlClientFailoverConfigBuilder() {
+        this((YamlClientFailoverConfigLocator) null);
+    }
+
+    /**
+     * Constructs a {@link YamlClientFailoverConfigBuilder} that loads the configuration
+     * with the provided {@link YamlClientFailoverConfigLocator}.
+     * <p/>
+     * If the provided {@link YamlClientFailoverConfigLocator} is {@code null}, a new
+     * instance is created and the config is located in every possible
+     * places. For these places, please see {@link YamlClientFailoverConfigLocator}.
+     * <p/>
+     * If the provided {@link YamlClientFailoverConfigLocator} is not {@code null}, it
+     * is expected that it already located the configuration YAML to load
+     * from. No further attempt to locate the configuration YAML is made
+     * if the configuration YAML is not located already.
+     *
+     * @param locator the configured locator to use
+     * @throws HazelcastException if no failover configuration is found
+     */
+    public YamlClientFailoverConfigBuilder(YamlClientFailoverConfigLocator locator) {
+        if (locator == null) {
+            locator = new YamlClientFailoverConfigLocator();
+            locator.locateEverywhere();
+        }
+
+        if (!locator.isConfigPresent()) {
+            throw new HazelcastException("Failed to load ClientFailoverConfig");
+        }
+
+        this.in = locator.getIn();
+    }
+
+    public ClientFailoverConfig build() {
+        ClientFailoverConfig clientFailoverConfig = new ClientFailoverConfig();
+        build(clientFailoverConfig);
+        return clientFailoverConfig;
+    }
+
+    public void setProperties(Properties properties) {
+        setPropertiesInternal(properties);
+    }
+
+    void build(ClientFailoverConfig clientFailoverConfig) {
+        try {
+            parseAndBuildConfig(clientFailoverConfig);
+        } catch (Exception e) {
+            throw ExceptionUtil.rethrow(e);
+        } finally {
+            IOUtil.closeResource(in);
+        }
+    }
+
+    private void parseAndBuildConfig(ClientFailoverConfig config) throws Exception {
+        YamlMapping yamlRootNode;
+        try {
+            yamlRootNode = ((YamlMapping) YamlLoader.load(in));
+        } catch (Exception ex) {
+            throw new InvalidConfigurationException("Invalid YAML configuration", ex);
+        }
+
+        String configRoot = getConfigRoot();
+        YamlNode clientFailoverRoot = yamlRootNode.childAsMapping(configRoot);
+        if (clientFailoverRoot == null) {
```

Comment:
Can be replaced with Preconditions.checkTrue() as you did for other places.

Reply:
Done